src/main/java/com/es/phoneshop/model/product/Product.java

```diff
@@ -82,4 +98,40 @@ public String getImageUrl() {
     public void setImageUrl(String imageUrl) {
         this.imageUrl = imageUrl;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Product product = (Product) o;
+        return id.equals(product.id)
+                && code.equals(product.code)
+                && description.equals(product.description)
+                && price.equals(product.price)
+                && currency.equals(product.currency)
+                && stock == product.stock
+                && imageUrl.equals(product.imageUrl);
+    }
```

Comment:
This implementation may throw NPE, because class members do not guarantee to be non-null. Besides that, i don't think you need custom implementation for it and methods below - default implementation will work well for now.